engine/core/src/main/java/org/datacleaner/metamodel/datahub/DataHubDataContext.java

```diff
@@ -166,8 +174,43 @@ public void executeGoldenRecordDelete(String goldenRecordId) {
     public void executeSourceRecordDelete(String source, String id, String recordType) {
         String uri = _updateConnection.getDeleteSourceRecordUrl();
         uri = uri + "/" + source + "/" + id + "/" + recordType;
-        logger.debug("request {}", uri);
+        LOGGER.debug("request {}", uri);
         final HttpDelete request = new HttpDelete(uri);
         executeRequest(request, _updateConnection.getHttpClient());        
     }
+    
+    private static class UserInfo {
+        @SuppressWarnings("unused")
+        public String username;
+        public String tenant;
+    }
+
+    private String retrieveTenantName() {
+        final String getUserInfoUrl = _repoConnection.getUserInfoUrl();
+        final HttpGet request = new HttpGet(getUserInfoUrl);
+        try (final MonitorHttpClient monitorHttpClient = _repoConnection.getHttpClient()) {
+            final HttpResponse response = monitorHttpClient.execute(request);
+
+            final StatusLine statusLine = response.getStatusLine();
+
+            if (statusLine.getStatusCode() == HttpStatus.SC_OK
+                    || statusLine.getStatusCode() == HttpStatus.SC_CREATED) {
+                // read response as JSON.
+                final InputStream content = response.getEntity().getContent();
+                final UserInfo userInfo;
+                try {
+                    userInfo = new ObjectMapper().readValue(content, UserInfo.class);
+                    return userInfo.tenant;
+                } finally {
+                    FileHelper.safeClose(content);
+                }
+            } else {
+                final String reasonPhrase = statusLine.getReasonPhrase();
+                throw new RuntimeException("Failed to retrieve the tenant name: " + reasonPhrase);
+            }
+        } catch (Exception e) {
```

Comment:
Minor issue: This will also catch the RuntimeException thrown just above, so you'll have an additional level of exception causes to unravel. It would probably be good to check here if it is instanceof RuntimeException and if so, rethrow it instead of wrapping it.